compiler_opt/rl/generate_bc_trajectories.py

```diff
@@ -14,12 +14,101 @@
 # limitations under the License.
 """Module for running compilation and collect data for behavior cloning."""
 
-from typing import Callable, Dict, List, Optional
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type
+
+from absl import logging
+import os
+import shutil
 
 import numpy as np
 import tensorflow as tf
 from tf_agents.trajectories import policy_step
 from tf_agents.trajectories import time_step
+from tf_agents.specs import tensor_spec
+
+from compiler_opt.distributed import worker
+from compiler_opt.rl import corpus
+from compiler_opt.rl import env
+
+
+def add_int_feature(
+    sequence_example: tf.train.SequenceExample,
+    feature_value: np.int64,
+    feature_name: str,
+):
+  """Add an int feature to feature list.
+
+  Args:
+    sequence_example: sequence example to use instead of the one belonging to
+      the instance
+    feature_value: tf.int64 value of feature
+    feature_name: name of feature
+  """
+  f = sequence_example.feature_lists.feature_list[feature_name].feature.add()
+  lst = f.int64_list.value
+  lst.extend([feature_value])
+
+
+def add_float_feature(
+    sequence_example: tf.train.SequenceExample,
+    feature_value: np.float32,
+    feature_name: str,
+):
+  """Add a float feature to feature list.
+
+  Args:
+    sequence_example: sequence example to use instead of the one belonging to
+      the instance
+    feature_value: tf.int64 value of feature
+    feature_name: name of feature
+  """
+  f = sequence_example.feature_lists.feature_list[feature_name].feature.add()
+  lst = f.float_list.value
+  lst.extend([feature_value])
+
+
+def add_string_feature(
```

Comment:
why, do we ever think we'd use this?

Reply:
Imagine that during training we want to have specific weights only specific modules, then we can read off the module name from here and compute the relevant weight. It's something that we experimented with, however, it did not significantly improve on our final method. I can imagine this being useful in similar contexts though.

Reply:
if we don't currently (as in, part of this patch or known upcoming ones) use it, let's not add it. It's easy to add later.

Reply:
I have used it in part of the training for some runs just not the ones that ended up giving the best policy. It would be more effort to remove this and the module_name feature than actually keep it at this point but really up to you.

Reply:
ok! just wanted to make sure it has an envisioned use (other than test)